### ISSUE TITLE
508-defect-2 [SCREENREADER]: On 1 of 3 Veteran information, the button to send to Permanent address SHOULD be a radio input

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#9940]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

On step 1 of 3 Veteran information, Shipping address, the Permanent address selection **should** be a radio input rather than a button, in the semantic structure. The semantic options here are a choice between the Permanent and Temporary addresses. The default if there is no temporary address would be for the permanent address to be selected. 

The screen reader reads the current implementation as a button. A button is used to either submit a form or toggle a script. Without sight, when the screen reader reads that this is a button, one would expect to move to the next stage of the form. When nothing happens, it is confusing, and the user doesn't know what to do.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to understand that I am making a selection between address choices.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: staging

## Steps to Recreate

1. Enter https://staging.va.gov/hearing-aid-batteries-and-accessories/veteran-information/addresses in browser, logged in as 228 (or likely any other user)
2. Start screenreading device listed in Environment
3. Using only your ears and the keyboard, attempt to select the Permanent address on record to move through the form
4. Verify the experience is unclear

## Possible Fixes (optional)

Recommend that the Permanent address and the Temporary address are radio inputs that a user chooses between.

```html
<form>
  <fieldset>
    <legend>Permanent address</legend>
    <ul>
      <li>5554 Caithness Court</li>
      <li>APO, AZ 22032</li>
      <li>United States</li>
      <li><button>Edit permanent address</button></li>
    </ul>
    <label for="permanent_address"><input type="radio" id="permanent_address" checked /> Send to my permanent address</label>
  </fieldset>

  <fieldset>
    <legend>Temporary address</legend>
    <p>
      Please provide a temporary address if you want us to ship your order to another location, like a relative's house or a vacation home.
    </p>
    <p><button>Add a temporary address</button>
    <label for="temporary_address"><input type="radio" id="temporary_address" /> Send to my temporary address</label>  
    <!-- this radio input would likely be visually hidden, unless a temporary address is added -->
  </fieldset>
```

## WCAG or Vendor Guidance (optional)

* [CSS Tricks - A complete guide to links and buttons](https://css-tricks.com/a-complete-guide-to-links-and-buttons/)

## Screenshots or Trace Logs

![Screen Shot 2020-06-08 at 10 50 26 AM](https://user-images.githubusercontent.com/57469/84044703-e830c700-a975-11ea-8cc3-b2f6ce42b564.png)

If a user has addresses on file, this is the screen they see. This is more like what the screen when there is only a permanent address on file should look like, but with the empty temporary radio input hidden.

![Screen Shot 2020-06-08 at 12 41 28 PM](https://user-images.githubusercontent.com/57469/84057363-8f692a80-a985-11ea-88f6-ac307946931b.png)

**Comment by mr0sari0** [comment]
Hi @jenstrickland, for the case with only a permanent address on file,  I think our team thought it wouldn't make sense to present a radio button for one choice (the permanent address), until a temporary address is added and then both choices would be radio buttons.

For the cases where there is a permanent address and temporary address on file, both choices are radio buttons.

Are you saying that we should make the case where there is only a permanent address on file be a radio button?

**Comment by jenstrickland** [comment]
The options here are:
- Send to permanent address on file
- Add temporary delivery address

The radio input options could be visually presented as buttons. The Add temporary delivery address button would reveal the edit in place form to add the temporary address.

@rebeccawalsh — does this recommendation make sense, or are there design reasons not to?

**Comment by rebeccawalsh** [comment]
@jenstrickland @mr0sari0 

That may work, but it's worth noting that selecting the second "Add temporary delivery address" radio button would immediately cause the button to disappear and open the Add Address state on that card. They should be able to cancel to go back. But what happens to the radio button if they cancel that 'Add Address' state? Does it automatically select permanent address again? This might feel weird. 

Do you think these labels would be as effective?
- Send my order to this address
- Send my order to a temporary address

**Comment by jenstrickland** [comment]
Using a number list here to clarify my own thinking:

1. The second option should be along the lines of "Add temporary delivery address" — to be clear what the action will be. 

1. That button would remain visible, in active state, as it opens the edit-in-place form to add the temporary address. 

1. The edit-in-place form would include the option to cancel, and if selected, the edit-in-place form closes, then "Send to permanent address" would be checked, and focus returned to that option.

1. Once the edit-in-place form complete and the temporary address saved, focus would return to the button, with asynchronously updated label text "Send to temporary address."


How does that sound?

**Comment by rileyorr** [comment]
Hi everyone!! Either @mr0sari0 or @bradl3yC brought up this single radio button issue to me a while ago, and I asked if it would be possible to label the "Add a temporary address" button link as a radio button similar to @jenstrickland's proposed solution. If I remember correctly, there was a technical reason within the forms system and/or schema that prevented us from being able to do that. Mahariel and/or Brad, do either of you remember why it wasn't possible?

**Comment by bradl3yC** [comment]
@rileyorr you're correct - I thought we had switched these a while back or it was documented somewhere that this needed to be done - I guess it wasn't :(

**Comment by rileyorr** [comment]
@bradl3yC It's all good - things happen! We've been working on a lot of updates lately so it's been challenging to keep track of everything. Do you think this work is accomplishable before launch?

**Comment by bradl3yC** [comment]
@rileyorr what can we put in there if not a radio button? The solution is to not have a radio button to keep in line with html5 spec

**Comment by jenstrickland** [comment]
In my comment above I outlined having two radio inputs which would be web standards compliant.

Is there something in the forms system that prevents doing this?

**Comment by bradl3yC** [comment]
@jenstrickland if I understand you correctly, the "Add a temporary address" link would be a radio input?

**Comment by jenstrickland** [comment]
Yes, I'll adjust my [comment above](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9940#issuecomment-642786172) for clarity. 😬

**Comment by mr0sari0** [comment]
Hi @jenstrickland, these changes should be in staging now.

**Comment by jenstrickland** [comment]
Validated. 🎉
